Template10 (Library)/Services/NavigationService/NavigationLogic.cs

```diff
@@ -171,6 +171,10 @@ public void SetupViewModel(INavigationService service, ITemplate10ViewModel view
                 var vm = viewmodel as Portable.IConfirmNavigationAsync;
                 return !await vm?.CanNavigateAsync(parameters);
             }
+            else if (viewmodel != null)
```

Comment:
Isn't non navigating better if the DataContext is not the expected ViewModel?
This is an edge-case I came across while changing from view page to view-model and thought a good idea to catch this out. If someone (like I did!) is messing up with a non-compliant datacontext it's better not to navigate.

What would you do with a  different type of datacontext? perhaps throw an exception? It's very rare but not impossible. From programming point of view, the if-condition exists and something needs to be done rather than leave it as unknown.